static/js/recent_topics_ui.js

```diff
@@ -147,25 +184,34 @@ function set_table_focus(row, col, using_keyboard) {
         }
     }
 
-    const message = {
-        stream: $topic_row.find(".recent_topic_stream a").text(),
-        topic: $topic_row.find(".recent_topic_name a").text(),
-    };
+    const type = get_row_type(row);
+    let message;
+    if (type === "private") {
+        message = {
+            recipients: $topic_row.find(".recent_topic_name a").text(),
+        };
+    } else {
+        message = {
+            stream: $topic_row.find(".recent_topic_stream a").text(),
+            topic: $topic_row.find(".recent_topic_name a").text(),
+        };
+    }
     compose_closed_ui.update_reply_recipient_label(message);
     return true;
 }
 
 export function get_focused_row_message() {
+    const recent_conversation_id_prefix_len = "recent_conversation:".length;
```

Comment:
I'd define `const recent_conversation_key_prefix = "recent_conversation:"` at the top of the file, and use that variable both in the `.slice` below and in the related code that generates this prefix.